packages/nimble-components/src/wafer-map/modules/zoom-handler.ts

```diff
@@ -0,0 +1,123 @@
+import { select } from 'd3-selection';
+import {
+    zoom,
+    ZoomBehavior,
+    zoomIdentity,
+    ZoomTransform,
+    zoomTransform
+} from 'd3-zoom';
+import type { DataManager } from './data-manager';
+import type { RenderingModule } from './rendering';
+
+/**
+ * InteractionHandler deals with user interactions and events like zooming
+ */
+export class ZoomHandler {
+    private zoomTransform: ZoomTransform = zoomIdentity;
+    private readonly canvas: HTMLCanvasElement;
+    private readonly zoomContainer: HTMLElement;
+    private readonly dataManager: DataManager;
+    private readonly renderingModule: RenderingModule;
+
+    public constructor(
+        canvas: HTMLCanvasElement,
+        zoomContainer: HTMLElement,
+        dataManager: DataManager,
+        renderingModule: RenderingModule
+    ) {
+        this.zoomContainer = zoomContainer;
+        this.canvas = canvas;
+        this.dataManager = dataManager;
+        this.renderingModule = renderingModule;
+    }
+
+    public createZoomBehavior(): ZoomBehavior<Element, unknown> {
+        this.canvas.addEventListener('wheel', event => event.preventDefault());
+        if (this.dataManager === undefined) return zoom();
+        const zoomBehavior = zoom()
+            .scaleExtent([
+                1.1,
+                this.getZoomMax(
+                    this.canvas.width * this.canvas.height,
+                    this.dataManager.containerDimensions.width * this.dataManager.containerDimensions.height
+                )
+            ])
+            .translateExtent([
+                [-100, -100],
```

Comment:
Could you create a constant for the `100` value whose name explains its meaning? I'm not clear what it means but maybe something like `canvasOffset` or `canvasPadding`?

Reply:
Same for the `1.1` value used for a few comparisons with `tranform.k`.